### ISSUE TITLE
Better handling of mocks for service tests

### ISSUE THREAD

**Issue by sp35** [intelowlproject/go-intelowl#37]
Better handling of mocks for service tests

_Originally posted by @sp35 in https://github.com/intelowlproject/go-intelowl/pull/35#discussion_r922804018_

**Comment by sp35** [comment]
Improvements that can be made:

- [ ] Mocking the test server: right now test server is being initiated with expected response for each test case.
https://github.com/intelowlproject/go-intelowl/blob/18821f3e29ffaf89a89b465e0e903fce6fb313f0/tests/jobService_test.go#L32-L33
- [ ] Mock handlers which take care of generating mocks for each resource
```golang
func mockedJobResponse(w http.ResponseWriter, r *http.Request)
func mockedJobListResponse(w http.ResponseWriter, r *http.Request)
```
- [ ] Non-success test cases (check if the client is handling errors in the expected way)

**Comment by fear-the-reaper** [comment]
@sp35 Non-success test cases are being checked wherever possible. Also we can use `TestMain` to instantiate the test server and use [this](https://clavinjune.dev/en/blogs/mocking-http-call-in-golang-a-better-way/) as a reference. One concern is the switch case can become too large.

**Comment by fear-the-reaper** [comment]
Implemented with PR: #56
